File: branches/SUSD/Minerva/Core/Viz/Progress.cpp

///////////////////////////////////////////////////////////////////////////////
//
//  Copyright (c) 2007, Arizona State University
//  All rights reserved.
//  BSD License: http://www.opensource.org/licenses/bsd-license.html
//  Created by: Adam Kubach
//
///////////////////////////////////////////////////////////////////////////////

#include "Minerva/Core/Viz/Progress.h"

#include <iostream>

using namespace Minerva::Core::Viz;

USUL_IMPLEMENT_IUNKNOWN_MEMBERS( Progress, Progress::BaseClass );

Progress::Progress() : BaseClass()
{
}

Progress::~Progress()
{
}

// Show the progress bar
void Progress::showProgressBar()
{
}
	
// Set the total of progress bar
void Progress::setTotalProgressBar ( unsigned int value )
{
}

// Update the progress bar
void Progress::updateProgressBar ( unsigned int value )
{
  if( value % 5 == 0 )
    std::cerr << value << std::endl;
}

// Hide the progress bar
void Progress::hideProgressBar()
{
}

Usul::Interfaces::IUnknown * Progress::queryInterface ( unsigned long iid )
{
  switch ( iid )
  {
  case Usul::Interfaces::IUnknown::IID:
  case Usul::Interfaces::IProgressBar::IID:
    return static_cast < Usul::Interfaces::IProgressBar* > ( this );
  }

  return 0x0;
}

File: Legacy/DotNet/Tools/Progress.cs

///////////////////////////////////////////////////////////////////////////////
//
//  Copyright (c) 2006, Perry L Miller IV
//  All rights reserved.
//  BSD License: http://www.opensource.org/licenses/bsd-license.html
//
///////////////////////////////////////////////////////////////////////////////

namespace CadKit.Tools
{
  public static class Progress
  {
    /// <summary>
    /// Report progress if possible.
    /// </summary>
    public static void notify(string text, int value, int total, object caller)
    {
      lock ("CadKit.Tools.Progress.notify")
      {
        CadKit.Interfaces.IProgressBar progress = caller as CadKit.Interfaces.IProgressBar;
        if (null != progress)
        {
          progress.Text = (null != text) ? text : "";
          float v = value;
          float t = total;
          float fraction = v / t;
          float position = fraction * progress.Range;
          progress.Value = (int)position;
        }

        CadKit.Interfaces.IUpdateDisplay update = caller as CadKit.Interfaces.IUpdateDisplay;
        if (null != update)
        {
          update.updateDisplay();
        }
      }
    }
  }
}
